### PR TITLE
UI component for Editing config items

### LINKED ISSUE #204: [Feature Request] - UI Component for Editing Config Items
### Pre-requisites:
- [x] I looked through both [open and closed issues](../issues?utf8=✓&q=is%3Aissue) and did not find another request for the same or similar feature.

## Is your feature request related to a problem, or an open issue? Describe the problem, and if applicable, reference the issue.

Currently there isn't an ability to change config items on the front-end side of things.

## Describe the solution you'd like

As it stands, the user has to go into the database in order to change the config items. We want to implement a way for the user to interact with and change these config items on a page.

## Describe alternatives you've considered

This implementation can be added in a few ways. One idea is to hook each config item to a button and have the button serve as a toggle (we will also have a clear way to show which toggle option it is set to). Another way is to have a drop-down menu for each config item. There are multiple ways we could do this, so these are simply idea pitches. Additionally, we need to figure out which page this should be added to (admin page? Its own page? etc.).

### PR DESCRIPTION
**Pull request by oscargoddard**
Co-Authored-By: Dhruv Tailor <11343998+TheDivexz@users.noreply.github.com>
Co-Authored-By: Joseph Nalepka <70769982+Peruggia29@users.noreply.github.com>

## Pre-Requisites
- [X] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [X] Yes, I included and/or modified tests to cover relevent code **OR** my change is non-technical
- [X] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

## Description of Changes

created a button at the admin page
created a function to toggle the key

## Related Issues

closes #204

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -86,6 +95,9 @@ const AdminPage: NextComponentType = () => {
                     {configItem.key}
                   </label>
                   <input type="text" className="form-control" id={configItem.key} placeholder="Team name" value={configItem.value} disabled />
+                  <button type="button" className="btn btn-primary mb-2 mt-3" onClick={handleChange}>
+                    Change
+                  </button>
```
Comment (johnkahn) [inline src/pages/index.tsx:100]: This is where you'll need to pass the key and new value to your `handleChange` function. I would also change the button text to reflect exactly what it's doing.

```suggestion
                  <button type="button" className="btn btn-primary mb-2 mt-3" onClick={() => handleChange(configItem)}>
                    {configItem.value === 'true' ? 'Disable' : 'Enable'}
                  </button>
```

File: src/pages/index.tsx [hunk 2]
```diff
@@ -62,6 +61,16 @@ const AdminPage: NextComponentType = () => {
...
 
+  const handleChange = () => async (): Promise<void> => {
+    await fetch('/api/config', {
+      method: 'POST',
+      body: JSON.stringify({}),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+  };
```
Comment (johnkahn) [inline src/pages/index.tsx:72]: This endpoint requires that `configKey` and `configValue` are in the body. So you'll need to modify this `handleChange` function to take a parameter and pass those in.
```suggestion
  const handleChange = (configItem: Config) => async (): Promise<void> => {
    const newValue = configItem.value === 'true' ? 'false' : 'true';

    await fetch('/api/config', {
      method: 'POST',
      body: JSON.stringify({
        configKey: configItem.key,
        configValue: newValue
      }),
      headers: {
        'Content-Type': 'application/json',
      },
    });
  };
```

**Review by johnkahn — CHANGES_REQUESTED** [review]
Currently the POST call you are making isn't sending the correct values, that call will fail without them. Also can you put a screenshot of what it looks like?

**Comment by oscargoddard** [comment]
Screen Shot 2020-12-01 at 14 16 40 PM

**Pull request closed without merging**